src/jit/lclvars.cpp

```diff
@@ -3363,15 +3363,20 @@ void Compiler::lvaAllocOutgoingArgSpace()
         lvaTable[lvaOutgoingArgSpaceVar].lvRefCnt     = 1;
         lvaTable[lvaOutgoingArgSpaceVar].lvRefCntWtd  = BB_UNITY_WEIGHT;
 
-#if defined(PROFILING_SUPPORTED) && defined(_TARGET_AMD64_) && !defined(UNIX_AMD64_ABI) // No 4 slots for outgoing params on System V.
+        if (compUsesThrowHelper && (lvaOutgoingArgSpaceSize == 0))
+        {
+            // For Windows Amd64 we guarantee that if there is a call we allocate the
+            // ABI mandated 4 slots outgoing arg space.
+            lvaOutgoingArgSpaceSize = MIN_ARG_AREA_FOR_CALL;
+        }
+
         // If we are generating profiling Enter/Leave/TailCall hooks, make sure
         // that outgoing arg space size is minimum 4 slots.  This will ensure
         // that even methods without any calls will have 4-slot outgoing arg area.
         if (compIsProfilerHookNeeded() && (lvaOutgoingArgSpaceSize == 0))
         {
-            lvaOutgoingArgSpaceSize = 4 * REGSIZE_BYTES;            
+            lvaOutgoingArgSpaceSize = MIN_ARG_AREA_FOR_CALL;
```

Comment:
condition (lvaOutgoingArgSpaceSize == 0) got repeated plus it is sufficient to check only one of the conditions to set lvaOutgoingArgSpaceSize.  I would suggest to rewrite this as follows:

if (lvaOutgoingArgSpaceSize  == 0)
{
     // comment lines go here
     if (compUsesThrowHelper  || compIsProfilerHookNeeded())
         lvaOutgoingArgSpaceSize = MIN_ARG_AREA_FOR_CALL;
}